### PR TITLE
Urgent wish action space : birth

### PR DESCRIPTION
**Pull request by yangmenghan**
This PR implement the birth part of `urgent wish for children` action:
- Create a new dwarf for the player
- Place it on the action space
- Make the new dwarf busy

PS: not sure that the description above is totally useful since it repeats information provided by commits alreayd 🤔 wdyt?

### CODE REVIEW

File: domain/urgentWishForChildren.unit.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+describe("Urgent Wish for Children", () => {
+    shouldPlaceDwarf(UrgentWishForChildren.execute);
+
+    it("should add new dwarf to action space", function () {});
+
+    it("new dwarf should be busy", function () {});
+
+    it("should pay for dwelling", function () {});
+
+    it("should add dwelling to player's store", function () {});
```
Comment (yangmenghan) [inline domain/urgentWishForChildren.unit.ts:89]: J'aurais pe pas du ajouter des tests vide ici, comme la liste n'est pas exhaustive, mais c'est cool d'avoir une todo list 🤔  wdyt?

File: domain/entity/Furnishing.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Resources } from "./Resources";
+
+export class Furnishing {
```
Comment (mdallongeville) [inline domain/entity/Furnishing.ts:3]: I don't remember the details, we want to separate Furnishing from just Tiles?

File: domain/entity/ActionSpace.ts [hunk 3]
```diff
@@ -4,23 +4,32 @@ import { Game } from "./Game";
 import { Resources } from "./Resources";
 
 export class ActionSpace {
-    constructor(id: ActionSpaceId, action: Action, replenishment?: Replenishment, dwarf?: Dwarf) {
+    constructor(
+        id: ActionSpaceId,
+        action: Action,
+        replenishment?: Replenishment,
+        dwarf?: Dwarf,
+        newBornDwarf?: Dwarf
```
Comment (mdallongeville) [inline domain/entity/ActionSpace.ts:12]: So we'll be using this to know that this dwarf should only be fed 1 PN?

(Or maybe we'll think about that later when we'll actually need this? ^^)

File: domain/urgentWishForChildren.unit.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                (mutation) => !!mutation.diff.newBornDwarf
+            );
+        });
+
+        it("new dwarf should be the same for player and action space", function () {
+            const { game, player } = buildBaseObjects();
+
+            const mutations = UrgentWishForChildren.execute(game, player.id);
+
+            const actionSpaceMutation = mutations
```
Comment (mdallongeville) [inline domain/urgentWishForChildren.unit.ts:38]: Detail: I would have created 2 sub-functions here `getActionSpaceDwarf()` and `getPlayerDwarf()` inside the `it`

(but feel free to keep it like that, it doesn't matter much)

**Review by mdallongeville — APPROVED** [review]
Good job!

File: domain/actionUtils.ts [hunk 5]
```diff
@@ -24,4 +24,14 @@ export namespace ActionUtils {
             { original: player, diff: { resources: player.resources.add(actionSpace.resources) } },
         ];
     }
+
+    export function giveBirthToDwarf(
+        actionSpace: ActionSpace,
+        player: Player
+    ): EntityMutation<EntityType>[] {
+        const newDwarf = new Dwarf();
+        const newPlayerDwarfs = new Map(player.dwarfs);
```
[the remark this answers is not in the document]
Comment (yangmenghan) [reply domain/actionUtils.ts:34]: it seems yes 🤔 : https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Map
(Syntaxe section, as map is a iterable of pairs)

File: domain/entity/Furnishing.ts (hunk 2 above)
Comment (yangmenghan) [reply domain/entity/Furnishing.ts:3]: Good question. Indeed I didn't checked the old plan. wdyt about having `Tile` as parent class of Furnishing?

File: domain/entity/ActionSpace.ts (hunk 3 above)
Comment (yangmenghan) [reply domain/entity/ActionSpace.ts:12]: Hmmm I'm rather for the option "Or maybe we'll think about that later when we'll actually need this?" for now 😬 
Having sth like `isBaby` in dwarf might be more clean 🤔

File: domain/urgentWishForChildren.unit.ts (hunk 4 above)
Comment (yangmenghan) [reply domain/urgentWishForChildren.unit.ts:45]: Agree, it makes the test more understandable. changed!

File: domain/entity/Furnishing.ts (hunk 2 above)
Comment (mdallongeville) [reply domain/entity/Furnishing.ts:3]: Hmm sure why not, we'll see

File: domain/urgentWishForChildren.unit.ts [hunk 6]
```diff
@@ -35,20 +35,26 @@ describe("Urgent Wish for Children", () => {
 
             const mutations = UrgentWishForChildren.execute(game, player.id);
 
+            const actionSpaceDwarf = getActionSpaceNewDwarf(mutations);
+            const playerDwarf = getPlayerNewDwarf(mutations, player);
+
+            expect(actionSpaceDwarf).to.deep.equal(playerDwarf);
```
Comment (mdallongeville) [inline domain/urgentWishForChildren.unit.ts:41]: @yangmenghan [Detail] You could also write it as `expect(getActionSpaceNewDwarf(mutations)).to.deep.equal(getPlayerNewDwarf(mutations, player));` if you prefer

But that's debatable, I let you see

**Pull request merged**